ledger/partial/ptrie/partialTrie.go

```diff
@@ -122,41 +107,39 @@ func NewPSMT(
 				if currentNode.lChild == nil { // check left child
 					currentNode.lChild = newNode(v, currentNode.height-1)
 				}
-				//  else if !bytes.Equal(currentNode.lChild.ComputeValue(), v) {
-				// 	return nil, fmt.Errorf("incompatible proof (left node value doesn't match) expected [%x], got [%x]", currentNode.lChild.ComputeValue(), v)
-				// }
 				if currentNode.rChild == nil { // create the right child if not exist
+					// Caution: we are temporarily initializing the node with default hash, which will later get updated to the
+					// proper value (if this is an interim node, its hash will be set when computing the root hash of the PTrie
+					// in the end; if this is a leaf, we'll set the hash at the end of processing the proof)
 					currentNode.rChild = newNode(defaultHash, currentNode.height-1)
 				}
 				currentNode = currentNode.rChild
 			} else { // left branching
 				if currentNode.rChild == nil { // check right child
 					currentNode.rChild = newNode(v, currentNode.height-1)
 				}
-				// else if !bytes.Equal(currentNode.rChild.ComputeValue(), v) {
-				// 	return nil, fmt.Errorf("incompatible proof (right node value doesn't match) expected [%x], got [%x]", currentNode.rChild.ComputeValue(), v)
-				// }
 				if currentNode.lChild == nil { // create the left child if not exist
+					// Caution: we are temporarily initializing the node with default hash, which will later get updated to the
+					// proper value (if this is an interim node, its hash will be set when computing the root hash of the PTrie
+					// in the end; if this is a leaf, we'll set the hash at the end of processing the proof)
 					currentNode.lChild = newNode(defaultHash, currentNode.height-1)
 				}
 				currentNode = currentNode.lChild
 			}
 		}
 
 		currentNode.payload = payload
-		currentNode.path = path
-		// update node's hashvalue only for inclusion proofs (for others we assume default value)
+		// update node's hash value only for inclusion proofs (for others we assume default value)
 		if pr.Inclusion {
 			currentNode.hashValue = ledger.ComputeCompactValue(hash.Hash(path), payload.Value, currentNode.height)
 		}
 		// keep a reference to this node by path (for update purpose)
 		psmt.pathLookUp[path] = currentNode
-
 	}
 
 	// check if the rootHash matches the root node's hash value of the partial trie
-	if psmt.root.HashValue() != rootValue {
-		return nil, fmt.Errorf("rootNode hash doesn't match the proofs expected [%x], got [%x]", psmt.root.HashValue(), rootValue)
+	if ledger.RootHash(psmt.root.forceComputeHash()) != rootValue {
```

Comment:
Is `forceComputeHash()` the step that forces all hashes of interim nodes to get computed?

Reply:
yes: https://github.com/onflow/flow-go/blob/92363cce23bea38475566de195378a39255a1e7f/ledger/partial/ptrie/node.go#L41-L42